src/twisted/words/protocols/irc.py

```diff
@@ -2621,9 +2625,17 @@ def connectionMade(self):
             self.register(self.nickname)
 
     def dataReceived(self, data):
-        basic.LineReceiver.dataReceived(self, data.replace('\r', ''))
+        data = data.replace('\r', '')
+        if isinstance(data, unicode):
+            data = data.encode("utf-8")
+        basic.LineReceiver.dataReceived(self, data)
+
 
     def lineReceived(self, line):
+        if bytes != str and isinstance(line, bytes):
+            # decode bytes from transport to unicode
```

Comment:
My understanding is `bytes != str` is used to distinguish python2 and python3 but this doesn't look good better to use `_PY3`